src/network/social/loader.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file network_client.h Client part of the network protocol. */
+
+#ifndef NETWORK_SOCIAL_LOADER_H
+#define NETWORK_SOCIAL_LOADER_H
+
+#include "../../core/alloc_type.hpp"
+#include "social_api.h"
+
+struct SocialPlatformPlugin {
+	void* handle;
+
+	OTTD_Social_Initialize initialize;
+	OTTD_Social_Shutdown shutdown;
+	OTTD_Social_Dispatch dispatch;
+	OTTD_Social_NewState newState;
+
+	void* userdata;
+};
+
+class SocialPlatformLoader : public ZeroedMemoryAllocator {
```

Comment:
You don't _need_ to use ZeroedMemoryAllocator as base class, and I think it might actually be Undefined Behavior to use it with a class that includes non-trivially-constructible members such as a std::vector.